processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -231,14 +232,15 @@ public void run() {
                 commitManager.maybeCommitAsync();
                 metrics.commitOffsetTime.record(timer.duration());
             }
+            waitForRemainingTasksCompletion(
```

Comment:
I think at this point we should say the subscription's state is at "SHUTTING_DOWN", so this should goes the line below https://github.com/line/decaton/pull/79/files#diff-37fa89e1da134b048d335ebf4753a63374e56c59dd65882fc1e84ac1aad44cacR240 ?

Reply:
Agreed, though I think that means we have to duplicate that line between the regular and exception case. Or do you think we should `waitForRemainingTasksCompletion` in the case of an exception as well?

Reply:
That's true.

> we should waitForRemainingTasksCompletion in the case of an exception as well?

I don't think so, it depends on exception type but when an error happens it's unlikely that all tasks could complete.